File: jobs/scripts/transaction-file.py
import os
import mysql.connector
import json
from timeit import default_timer as timer

start = timer()

mydb = mysql.connector.connect(
  host=os.getenv('DB_URL'),
  user=os.getenv('DB_USER'),
  password=os.getenv('DB_PASSWORD'),
  database=os.getenv('DB_NAME')
)

transactions_file = open('jobs/batches/transactions.json')

transactions = json.load(transactions_file)

mycursor = mydb.cursor()
number_transactions = len(transactions)
number_processed = 0
for transaction in transactions:
    sql = f"UPDATE clients SET balance = balance {'+' if transaction['OperationType']=='IN' else '-'} {transaction['Amount']} WHERE code = {transaction['ClientID']}"

    mycursor.execute(sql)

    mydb.commit()

    number_processed += 1

end = timer()
elapsed = end - start

mycursor.close()
mydb.close()

print('::', (number_processed / number_transactions) * 100, '%', 'Completed in', str(elapsed) + 's')
  

File: jobs/scripts/transaction-file.js
const fs = require('fs');
const mysql = require('mysql2/promise');

(async () => {
    var start = performance.now();

    var conn = await mysql.createConnection({
      host: process.env.DB_URL,
      user: process.env.DB_USER,
      password: process.env.DB_PASSWORD,
      database: process.env.DB_NAME,
    });
    
    
    let transactions = JSON.parse(fs.readFileSync('jobs/batches/transactions.json'));
    
    let number_processed = 0;
    let number_transactions = transactions.length;

    for (const transaction of transactions) {
        await conn.execute(`UPDATE clients SET balance = balance ${transaction.OperationType === 'IN' ? '+' : '-'} ${transaction.Amount} WHERE code = ${transaction.ClientID}`);
        number_processed++;
    }

    var end = performance.now();
    var elapsed = end - start;

    console.log("::", (number_processed / number_transactions) * 100, "% Completed in", elapsed/1000 + 's');    
})();